python-lib/dku_stress_test_center/stress_test_center.py

```diff
@@ -0,0 +1,193 @@
+
+    clean_filter = stress_test_indicator == DkuStressTestCenterConstants.CLEAN
+    clean_y_true = y_true[clean_filter]
+    clean_y_pred = y_pred[clean_filter]
+    clean_accuracy = accuracy_score(clean_y_true, clean_y_pred)
+    clean_f1_score = f1_score(clean_y_true, clean_y_pred, pos_label=pos_label)
+
+    stress_ids = np.unique(stress_test_indicator)
+    stress_ids = np.delete(stress_ids, np.where(stress_ids==DkuStressTestCenterConstants.CLEAN))
+    for stress_id in stress_ids:
+        stress_filter = stress_test_indicator == stress_id
+        stress_y_true = y_true[stress_filter]
+        stress_y_pred = y_pred[stress_filter]
+
+        stress_accuracy = accuracy_score(stress_y_true, stress_y_pred)
+        stress_f1_score = f1_score(stress_y_true, stress_y_pred, pos_label=pos_label)
+
+        stress_acc_drop = stress_accuracy - clean_accuracy
+        stress_f1_drop = stress_f1_score - clean_f1_score
+
+        if stress_id == DkuStressTestCenterConstants.PRIOR_SHIFT:
+            robustness = balanced_accuracy_score(stress_y_true, stress_y_pred)
+        else:
+            # 1 - Attack Success Rate
+            clean_correctly_predicted_filter = clean_y_pred == clean_y_true
+            robustness = accuracy_score(clean_y_pred[clean_correctly_predicted_filter],
+                                        stress_y_pred[clean_correctly_predicted_filter])
+
+        metrics_df = metrics_df.append({
+            DkuStressTestCenterConstants.STRESS_TEST_TYPE: stress_id,
+            DkuStressTestCenterConstants.ACCURACY_DROP: stress_acc_drop,
+            DkuStressTestCenterConstants.F1_DROP: stress_f1_drop,
+            DkuStressTestCenterConstants.ROBUSTNESS: robustness
+        }, ignore_index=True)
+
+    return metrics_df
+
+
+def get_critical_samples(y_true_class_confidence: np.array,  # n_rows x 1
+                         stress_test_indicator: np.array,
+                         row_indicator: np.array,
+                         top_k_samples: int = 5):
+    # sparse format for the pair (orig_x, pert_x)
+
+    valid_stress_ids = set(stress_test_indicator.unique()) & set(
+        DkuStressTestCenterConstants.PERTURBATION_BASED_STRESS_TYPES)
+    valid_stress_ids |= set(DkuStressTestCenterConstants.CLEAN)
+
+    true_class_confidence_df = pd.DataFrame({
+        DkuStressTestCenterConstants.STRESS_TEST_TYPE: stress_test_indicator,
+        DkuStressTestCenterConstants.DKU_ROW_ID: row_indicator,
+        DkuStressTestCenterConstants.CONFIDENCE: y_true_class_confidence
+    })
+
+    true_class_confidence_df = true_class_confidence_df[
+        true_class_confidence_df[DkuStressTestCenterConstants.STRESS_TEST_TYPE].isin(valid_stress_ids)]
+
+    # critical samples evaluation
+    # sort by std of uncertainty
+
+    std_confidence_df = true_class_confidence_df.groupby([DkuStressTestCenterConstants.DKU_ROW_ID]).std()
+    uncertainty = std_confidence_df[DkuStressTestCenterConstants.CONFIDENCE]
```

Comment:
To deal with constant prediction (hence NaN std). If all NaN, this will return an empty Serie.
```suggestion
    uncertainty = std_confidence_df[DkuStressTestCenterConstants.CONFIDENCE].dropna()
```